task_manager/repair_task.hh

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager.hh"
+
+class repair_task : public task_manager::task {
```

Comment:
I'm having second thought about defining the module tasks as derived from `task_manager::task`.
If we'll have enough generic code in `task_manager::task`, It might be better to define a virtual `task_manager::task::impl` class and make define this as `class repair_task : public task_manager::task::impl`, so that the generic code could call the `impl` methods.

Reply:
Also, repair_task should be defined in the repair folder.
It should be opaque to the task_manager, and to other task modules.